charts/metering-ansible-operator/templates/crds/hive.crd.yaml

```diff
@@ -21,3 +21,216 @@ spec:
+                More info: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-StorageFormatsStorageFormatsRowFormat,StorageFormat,andSerDe
+              minLength: 1
+            tableProperties:
+              type: array
+              description: |
+                TableProperties is an array and allows you to tag the table definition with your
+                own metadata key/value pairs. Some predefined properties exist to control
+                behavior of the table as well.
+                Note: this is an optional field.
+                More info: https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-listTableProperties
+            external:
+              type: boolean
+              description: |
+                External controls whether an external table is created, instead of a managed table.
+                If external is set to true, this causes Hive to point to an existing location,
+                as specified by the "location" field.
+                Note: this is an optional field.
+            managePartitions:
+              type: boolean
+              description: |
+                ManagePartitions controls whether the reporting-operator needs to check
+                if the table partitions match the partitions listed in the "partitions" field.
+                Note: this is an optional field.
+            partitions:
+              type: array
+              description: |
+                A list of partitions that this Hive table should contain.
+                Note: this is an optional field.
+              nullable: true
+              items:
+                type: object
+                required:
+                - partitionSpec
+                - location
+                properties:
+                  partitionSpec:
+                    type: array
+                    description: |
+                      PartitionSpec is a map containing string keys and values, where each key
+                      is expected to be the name of a partition column, and the value is the
+                      value of the partition column.
+                  location:
+                    type: string
+                    description: |
+                      Location specifies where the data for this partition is stored.
+                      This should be a sub-directory of the "location" field.
+                    minLength: 1
+                    format: uri
+          oneOf:
+          - required:
+            - numBuckets
+            - clusteredBy
+          - required:
+            - external
+            - location
+            properties:
+              external:
+                enum:
+                - true
+          - required:
+            - managePartitions
+            - partitions
+            properties:
+              managePartitions:
+                enum:
+                - true
+          - allOf:
```

Comment:
This case doesn't make much sense to me, managePartitions has nothing to do with `numBuckets` or `clusteredBy`. 

I think if you're trying to handle numBuckets + clusteredBy being dependent on each other, you could have something like (updating the top case in this oneOf and removing this allOf):
```
- oneOf:
   - required:
      - numBuckets
      - clusteredBy
   - not:
        required:
        - numBuckets
        - clusteredBy
```

Reply:
I needed a case that allows the minimal configuration (`columns`, `tableName`, and `databaseName`) as a valid configuration, but also allows other fields like `external`, `rowFormat`, `fileFormat`, etc.

I wasn't sure what the best approach was to do something like this, while still listing fields that are dependent on each other.

Reply:
Also, any `HiveTable` resource that's implicitly created will have a `managePartitions: false` and `partitions: null` included in the spec (like in the case where a `PrometheusMetricsImporter` ReportDatasource.)

Reply:
@timflannagan1 then why not add a oneOf case that has `required` for columns, tableName, and databaseName then?

Reply:
I was thinking about that initially, but it's redundant to specify both those fields as required in the property definitions, and then include them as required fields in the `oneOf` clause.

Reply:
Can we have an empty object or something in the oneOf? or empty required?

Reply:
That could work, assuming that an invalid configuration would produce two valid alternatives, and the `oneOf` clause would error, as it only allows a single valid configuration.

Reply:
We can use `anyOf` then

Reply:
Or a combination of nesting an anyOf within the oneOf

Reply:
We would need to use `oneOf` as the top-level clause. I can experiment w/ different combinations of nested logical junctor's.